cloudfour/cloudfour.com-patterns
# Fix Comment story performance issue

## Overview

This PR fixes the Comment story performance issue that was causing the browser to freeze when viewing the "Docs" tab. The proposed fix is to generate random comments ahead of time instead of inline inside the `<Story>` elements.

> // Before, the makeComment() function was called inline inside each `<Story>` but
> // the randomness caused the stories to render hundreds or thousands of times.
> // The page would freeze up. Now, we create a random set of comments ahead of
> // time and just reference them in the stories.

Thanks for the idea, @calebeby. It worked! 🎉 


## Screenshots

Before, navigating to the "Docs" tab would freeze the browser to the point where you couldn't scroll in Firefox. Now, you can scroll through the page:

![after](https://user-images.githubusercontent.com/459757/190511756-b6e781d1-d652-40bd-8a9e-e3e8d5c80619.gif)

### Before

Before, there was a very long yellow JS block, over 10 seconds in the screenshot below:

<img width="1279" alt="Screen Shot 2022-09-15 at 2 32 04 PM" src="https://user-images.githubusercontent.com/459757/190512689-9ec1c329-4254-4ace-a909-51e1fb98747c.png">

### After

After, the yellow JS block is not 10+ seconds long:

<img width="1279" alt="Screen Shot 2022-09-15 at 2 33 01 PM" src="https://user-images.githubusercontent.com/459757/190512709-2e782136-a45f-43a9-8b83-a5932524d3f9.png">


## Testing

The bug was originally reported in Firefox but test in all browsers.

> If you [view the Comment component in Storybook](https://cloudfour-patterns.netlify.app/?path=/story/components-comment--single) and attempt to switch to Docs view, the page completely tanks performance and appears to never resolve (at least in desktop Firefox as of this writing).

1. Load up the "Canvas" tab: https://deploy-preview-2048--cloudfour-patterns.netlify.app/?path=/story/components-comment--single
1. Switch over to the "Docs" tab and confirm the page doesn't lock up
2. Reload the "Docs" tab view, and confirm the page doesn't lock up

---

- Closes #2003
